### PR TITLE
Update project type to be gce-project.

### PR DESCRIPTION
**Pull request by sawsa307**
* Update project type to `gce-project` as `ingress-project` does not exist in community clusters.

/assign @aojea 
/assign @swetharepakula

### CODE REVIEW

**Comment by ameukam** [comment]
For reference the pool that will be used: https://github.com/kubernetes/k8s.io/blob/5eb099ddf8c9d7413b75119d043d3588711be831/infra/gcp/terraform/k8s-infra-prow-build/prow-build/resources/test-pods/boskos-resources-configmap.yaml#L5-L127

File: hack/run-e2e-gce.sh [hunk 1]
```diff
@@ -157,7 +157,6 @@ EOF
   --env=GCE_ALPHA_FEATURES=NetworkEndpointGroup \
   --env=KUBE_GCE_ENABLE_IP_ALIASES=true \
   --extract=ci/latest \
-  --gcp-project="${GCP_PROJECT}" \
```
Comment (aojea) [inline hack/run-e2e-gce.sh:160]: you need this

**Comment by aojea** [comment]
I think we just need only the change in `hack/boskos.sh`

**Comment by sawsa307** [comment]
Updated. PTAL!

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**